conmon-rs/server/src/attach.rs

```diff
@@ -188,74 +195,92 @@ impl Attach {
         }
     }
 
-    async fn read_loop(mut read_half: OwnedReadHalf, tx: Sender<Vec<u8>>) -> Result<()> {
+    async fn read_loop(mut read_half: OwnedReadHalf, tx: Sender<Vec<u8>>, token: CancellationToken) -> Result<()> {
         loop {
             let mut buf = vec![0; Self::PACKET_BUF_SIZE];
-            match read_half.read(&mut buf).await {
-                Ok(n) if n > 0 => {
-                    if let Some(first_zero_idx) = buf.iter().position(|&x| x == 0) {
-                        buf.resize(first_zero_idx, 0);
+            select! {
```

Comment:
summoning @cgwalters

I know you linked something in the past about how neferious select statements can be. I can't frankly see a better way to handle this. If you have any tips or a way to absolve the risk of select, I would appreciate it :)

Reply:
Oh heh you tagged me earlier and I missed it.

I think this case is safe, but I'd still add a comment with a rationale for why.